File: backendworker/NHSOnline.Backend.MessagesApi/Areas/Messages/Models/SenderContext.cs
using System;
using MongoDB.Bson.Serialization.Attributes;

namespace NHSOnline.Backend.MessagesApi.Areas.Messages.Models
{
    public class SenderContext
    {
        [BsonElement]
        public string SupplierId { get; set; }

        [BsonElement]
        public string CommunicationId { get; set; }

        [BsonElement]
        public string TransmissionId { get; set; }

        [BsonElement]
        public DateTime? CommunicationCreatedDateTime { get; set; }

        [BsonElement]
        public string RequestReference { get; set; }

        [BsonElement]
        public string CampaignId { get; set; }

        [BsonElement]
        public string OdsCode { get; set; }

        [BsonElement]
        public string NhsNumber { get; set; }

        [BsonElement]
        public string NhsLoginId { get; set; }
    }
}

File: bddtests/src/main/kotlin/worker/models/messages/SenderContext.kt
package worker.models.messages

data class SenderContext(var supplierId: String? = null,
                         var communicationId: String? = null,
                         var transmissionId: String? = null,
                         var communicationCreatedDateTime: String? = null,
                         var requestReference: String? = null,
                         var campaignId: String? = null,
                         var odsCode: String? = null,
                         var nhsNumber: String? = null,
                         var nhsLoginId: String? = null)
